### PR TITLE
Add client package

### PR DESCRIPTION
**Pull request by sd2k**
This moves the client logic written in https://github.com/grafana/grafana-plugin-sdk-go/pull/740
to a new client subpackage in this repository.

It's based on #58 since it uses the new health check details.

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		return false, fmt.Errorf("unmarshal response: %w", err)
+	}
+	return details.OpenAIEnabled, nil
+}
+
+func (o *openAI) ChatCompletions(ctx context.Context, req openai.ChatCompletionRequest) (openai.ChatCompletionResponse, error) {
+	return o.client.CreateChatCompletion(ctx, req)
+}
+
+func (o *openAI) StreamChatCompletions(ctx context.Context, req openai.ChatCompletionRequest) (*openai.ChatCompletionStream, error) {
```
Comment (csmarchbanks) [inline pkg/client/client.go:74]: Should this be `ChatCompletionsStream` to match the return type a bit better/keep the functions together in docs?

Comment (sd2k) [reply pkg/client/client.go:74]: Mmm I named it this to match the JS library but can change it to match the go-openai library 👍

**Review by csmarchbanks — COMMENTED** [review]
My thought on this was to move it to a sub-module inside of this repo. That way the dependency tree would be very small for anyone wanting to import it, rather than the whole plugin. What would you think of that? Pretty much just have a `llmclient/` or `grafanallm/` folder with a new go.mod in it.

**Comment by sd2k** [comment]
That makes sense, I actually wanted to know whether Go would be smart enough to figure out that the subpackage didn't need all of the dependencies though. Looking at [this PR](https://github.com/grafana/grafana-plugin-sdk-go/pull/742/files) it only adds a few files to the go.sum, so maybe it's OK? Or maybe that's because all of our dependencies are already present 😂

**Comment by csmarchbanks** [comment]
I think it will typically be smart enough in basic cases - until a different package also needs one of those dependencies and it needs to try to figure out which version... especially if the versions don't match or are not semver. Maybe it has gotten even smarter than that though!

**Comment by sd2k** [comment]
👍 I'll move stuff around to a subdirectory/module so we don't run into any issues later.

File: llmclient/llmclient.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+// on the given Grafana instance.
+func NewOpenAI(grafanaURL, grafanaAPIKey string) OpenAI {
+	httpClient := &http.Client{}
+	return NewOpenAIWithClient(grafanaURL, grafanaAPIKey, httpClient)
+}
+
+// NewOpenAIWithClient creates a new OpenAI client talking to the Grafana LLM app installed
+// on the given Grafana instance, using the given HTTP client.
+func NewOpenAIWithClient(grafanaURL, grafanaAPIKey string, httpClient *http.Client) OpenAI {
+	url := strings.TrimRight(grafanaURL, "/") + "/api/plugins/grafana-llm-app/resources/openai/v1"
```
Comment (yoziru) [inline llmclient/llmclient.go:47]: Could we put part of this path in a const? `/api/plugins/grafana-llm-app`

File: llmclient/llmclient.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+type healthCheckResponse struct {
+	Details struct {
+		OpenAIEnabled bool `json:"openAI"`
+		VectorEnabled bool `json:"vector"`
+	} `json:"details"`
+}
+
+func (o *openAI) Enabled(ctx context.Context) (bool, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", o.grafanaURL+"/api/plugins/grafana-llm-app/health", nil)
```
Comment (yoziru) [inline llmclient/llmclient.go:68]: Same as above: Could we put part of this path in a const? `/api/plugins/grafana-llm-app`

File: llmclient/llmclient.go (hunk 2 above)
Comment (yoziru) [reply llmclient/llmclient.go:47]: Don't really love the hardcoding here, but loosely held opinion so good to merge as-is 🤷

**Review by yoziru — APPROVED** [review]
LGTM

My main thought is that it's very openai-only based, but that applies to the current state of the plugin anyway, so this is a problem for future us type of thing (out of scope here)

File: llmclient/llmclient.go (hunk 2 above)
Comment (sd2k) [reply llmclient/llmclient.go]: Done (including the `/resources` part) 👍

**Pull request merged**